src/toolbox/util.rs

```diff
@@ -0,0 +1,28 @@
+use curve25519_dalek::scalar::Scalar;
+use std::cmp::Ordering;
+
+/// Raise a Scalar to an arbitrary power.  This code is SUPER dumb: not efficient and not constant-time.
+pub fn pow(x: &Scalar, p: usize) -> Scalar {
+    if p == 0 {
+        return Scalar::one();
+    }
+
+    let mut res = x.clone();
+    for _ in 1..p {
+        res *= x;
+    }
+    res
+}
+
+/// Compare two Scalar values.
+pub fn compare_scalars(a: &Scalar, b: &Scalar) -> Ordering {
```

Comment:
Isn't there a comparison between scalars in the library? If not, maybe add some testing as well

Reply:
The library only offers us Eq and PartialEq, sadly.  That's a good idea, I'll add in tests for this.

Reply:
Good thing you caught this - the tests uncovered a major bug that just coincidentally wasn't triggered by our indirect tests.  Fixed!